docs/content/config/advanced/kubernetes.md

```diff
@@ -2,392 +2,374 @@
+```yaml
+---
+apiVersion: v1
+kind: PersistentVolumeClaim
+
+metadata:
+  name: data
+
+spec:
+  storageClassName: local-path
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
 
-!!! example
+### Service
 
-    ```yaml
-    kind: ConfigMap
-    apiVersion: v1
-    metadata:
-      name: mailserver.config
-      labels:
-        app: mailserver
-    data:
-      postfix-main.cf: |
-        smtpd_recipient_restrictions = permit_sasl_authenticated, permit_mynetworks, reject_unauth_destination, reject_unauth_pipelining, reject_invalid_helo_hostname, reject_non_fqdn_helo_hostname, reject_unknown_recipient_domain, reject_rbl_client zen.spamhaus.org, reject_rbl_client bl.spamcop.net
-    # ...
+A `Service` is required for getting the traffic to the pod itself. The service is somewhat crucial. Its configuration determines whether the original IP from the sender will be kept. [More about this further down below](#exposing-your-mail-server-to-the-outside-world).
 
-    ---
+The configuration you're seeing does keep the original IP, but you will not be able to scale this way. We have chosen to go this route in this case because we think most K8s users will only want to have one instance anyway, and users that need high availability know how to do it anyways.
 
-    kind: Deployment
-    apiVersion: extensions/v1beta1
-    metadata:
-      name: mailserver
-    # ...
-        volumeMounts:
-          - name: config
-            subPath: postfix-main.cf
-            mountPath: /tmp/docker-mailserver/postfix-main.cf
-            readOnly: true
-    ```
+With this setup, you want your load-balancer (e.g. MetalLB) to give this service an external, especially routable IP address.
 
-### External IPs Service
+```yaml
+---
+apiVersion: v1
+kind: Service
+
+metadata:
+  name: mailserver
+  labels:
+    app: mailserver
+
+spec:
+  type: LoadBalancer
+  externalTrafficPolicy: Local
+
+  selector:
+    app: mailserver
+
+  ports:
+    # Transfer
+    - name: transfer
+      port: 25
+      targetPort: transfer
+      protocol: TCP
+    # ESMTP with implicit TLS
+    - name: esmtp-implicit
+      port: 465
+      targetPort: esmtp-implicit
+      protocol: TCP
+    # ESMTP with explicit TLS (STARTTLS)
+    - name: esmtp-explicit
+      port: 587
+      targetPort: esmtp-explicit
+      protocol: TCP
+    # IMAPS with implicit TLS
+    - name: imap-implicit
+      port: 993
+      targetPort: imap-implicit
+      protocol: TCP
+
+```
 
-The simplest way is to expose mailserver as a [Service][k8s-network-service] with [external IPs][k8s-network-external-ip].
+### Deployments
 
-!!! example
+Last but not least, the `Deployment` becomes the most complex component. The deployment brings maximal security measures without compromising on ease of use.
```

Comment:
I find this a somewhat confusing sentence -- I'd suggest we just mention what the purpose of the Deployment object is rather than talking about security (which you can do, but a deployment's purpose is not to enforce security). Something like "Lastly, the Deployment is the most complex component. It instructs Kubernetes how to run the docker-mailserver container and how to apply your ConfigMaps and persisted storage. Additionally, we can set options to enforce runtime security here."

Reply:
I find your sentence better too. I will copy it :)